lib/omniauth/strategies/telegram.rb

```diff
@@ -81,14 +81,15 @@ def check_errors
       end
 
       def check_fields
-        FIELDS.all? { |f| request.params.include?(f) }
+        REQUIRED_FIELDS.all? { |f| request.params.include?(f) }
       end
-      
+
       def check_signature
         secret = OpenSSL::Digest::SHA256.digest(options[:bot_secret])
-        signature = HASH_FIELDS.map { |f| "%s=%s" % [f, request.params[f]] }.join("\n")
+        keys = (request.params.keys & SIGNATURE_FIELDS).sort
```

Comment:
> Data-check-string is a concatenation of all received fields, sorted in alphabetical order

https://core.telegram.org/widgets/login

Reply:
@Hawk73 is the documentation outdated? or where do the `locale` param etc come from?

Reply:
Please let me explain it. I add `locale` param to callback URL for saving current locale. So this param returns with telegrams params. That is why I added whitelist of params to check signature correctly.

Reply:
wouldn't that allow an attacker to pass arbitrary data via the callback url-params? i'm not super into the omniauth login widget, so please point it out if i'm thinking in the wrong direction here.